sklearn/metrics/_plot/confusion_matrix.py

```diff
@@ -83,22 +83,37 @@ def plot(self, include_values=True, cmap='viridis',
         n_classes = cm.shape[0]
         self.im_ = ax.imshow(cm, interpolation='nearest', cmap=cmap)
         self.text_ = None
-
         cmap_min, cmap_max = self.im_.cmap(0), self.im_.cmap(256)
 
         if include_values:
             self.text_ = np.empty_like(cm, dtype=object)
-            if values_format is None:
-                values_format = '.2g'
 
             # print text with appropriate color depending on background
             thresh = (cm.max() + cm.min()) / 2.0
+
             for i, j in product(range(n_classes), range(n_classes)):
+
                 color = cmap_max if cm[i, j] < thresh else cmap_min
-                self.text_[i, j] = ax.text(j, i,
-                                           format(cm[i, j], values_format),
-                                           ha="center", va="center",
-                                           color=color)
+
+                if values_format is None:
+                    if cm.dtype.kind == "f":
+                        values_format = '.2g'
+                    elif cm[i, j] < 1e7:
+                        values_format = 'd'
+                    else:
+                        values_format = '.2g'
+
+                    self.text_[i, j] = ax.text(
+                        j, i, format(cm[i, j], values_format),
+                        ha="center", va="center",
+                        color=color)
+                    values_format = None
```

Comment:
Instead of doing this trick, just use `fmt` instead of `values_format` and remove the else condition below which is redundant

Reply:
Not sure what you mean by this, declare a new variable 'fmt' and use that in the values_format place?

Reply:
something like

```py
fmt = values_format
if fmt is None:
	fmt = ...


# and then
... format(cm[i, j], fmt), ...
```

Reply:
Yea, that makes more sense, thanks.